src/io/fasta.rs

```diff
@@ -22,11 +22,17 @@ use std::collections;
 use std::fs;
 use std::path::Path;
 use std::convert::AsRef;
+use std::cmp::min;
 
 use csv;
 
 use utils::{TextSlice, Text};
 
+
+/// Maximum size of temporary buffer used for reading indexed FASTA files.
+const MAX_FASTA_BUFFER_SIZE: usize = 512;
```

Comment:
This is for files with very long lines, such as the default output from 'seqtk seq' without the -l option. The previous implementation would allocate a vector of size 'line_bytes' regardless of size of the requested interval, resulting in an allocation that could be orders of magnitude larger than what is required. Naturally this should not be the common case, as most FASTA files are properly formatted, but I think it is worthwhile to prevent unexpected behavior (unbounded allocations) in the uncommon case.